graal/src/main/java/io/micronaut/graal/reflect/GraalTypeElementVisitor.java

```diff
@@ -302,19 +302,19 @@ public final void finish(VisitorContext visitorContext) {
     }
 
     private void generateNativeImageProperties(VisitorContext visitorContext) {
-        List<Map> json;
+        List<Map<String, Object>> json;
 
         Optional<Path> projectDir = visitorContext.getProjectDir();
 
-        File userReflectJsonFile = projectDir
-                .map(projectPath -> Paths.get(projectPath.toString(), BASE_REFLECT_JSON).toFile())
+        Path userReflectJsonFile = projectDir
+                .map(projectPath -> projectPath.resolve(BASE_REFLECT_JSON))
                 .orElse(null);
 
-        if (userReflectJsonFile != null && userReflectJsonFile.exists()) {
-            try {
-                json = MAPPER.readValue(userReflectJsonFile, new TypeReference<List<Map>>() {
+        if (userReflectJsonFile != null && Files.exists(userReflectJsonFile)) {
```

Comment:
Do we have anything that tests this?  Any other projects?  I don't think this has changed the coverage, as it probably wasn't tested before.

Reply:
i dont know a way to build a test that would make using path only here behave any differently, unfortunately.